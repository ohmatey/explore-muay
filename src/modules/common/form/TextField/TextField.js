import React from 'react'
import PropTypes from 'prop-types'
import { Field } from 'react-final-form'
import MaterialTextField from '@material-ui/core/TextField'

export const TextInput = ({
  input
}) => (
  <MaterialTextField
    name={input.name}
    value={input.value}
    onChange={input.onChange}
  />
)

const TextField = ({
  name
}) => (
  <Field
    name={name}
    component={TextInput}
  />
)

TextField.propTypes = {
  name: PropTypes.string.isRequired
}

export default TextField
