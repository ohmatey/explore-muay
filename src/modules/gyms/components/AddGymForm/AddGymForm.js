import React from 'react'

import TextField from '../../../common/form/TextField'

const AddGymForm = ({
  handleSubmit
}) => (
  <form onSubmit={handleSubmit}>
    <TextField name='gymName' />

    <TextField name='gymOwner' />

    <button type='submit'>Submit</button>
  </form>
)

export default AddGymForm
