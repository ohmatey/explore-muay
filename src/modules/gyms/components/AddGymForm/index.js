import AddGymForm from './AddGymForm'

export default AddGymForm
