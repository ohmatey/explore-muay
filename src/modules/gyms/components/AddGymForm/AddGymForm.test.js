import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Form } from 'react-final-form'

import AddGymForm from './AddGymForm'
import TextField from '../../../common/form/TextField'

describe('<AddGymForm />', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <AddGymForm />
    )
  })

  it('should render with no errors', () => {
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render with gymName and gymOwner InputFields', () => {
    const handleSubmit = jest.fn()

    component.setProps({
      handleSubmit
    })

    expect(component.find('form').length).toEqual(1)
    expect(component.find('form').prop('onSubmit')).toEqual(handleSubmit)

    expect(component.find('button').length).toEqual(1)
    expect(component.find(TextField).length).toEqual(2)
  })

  it('should submit the form when submit button is clicked', () => {
    const handleSubmit = jest.fn()

    component.setProps({
      handleSubmit
    })

    component.find('form').simulate('submit')

    expect(handleSubmit).toHaveBeenCalled()
  })
})
