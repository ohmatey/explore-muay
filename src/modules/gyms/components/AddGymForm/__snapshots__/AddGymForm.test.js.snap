// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddGymForm /> should render with no errors 1`] = `
<form>
  <TextField
    name="gymName"
  />
  <TextField
    name="gymOwner"
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
